cpp/tests/test_buffer_resource.cpp

```diff
@@ -486,10 +465,17 @@ class BufferResourceCopyToTest : public BaseBufferResourceCopyTest,
         std::size_t const dest_offset
     ) {
         auto length = source->size;
-        auto bytes_written = source->copy_to(*dest, dest_offset, stream);
+        buffer_copy(
+            *dest,
+            *source,
+            source->size,
+            std::ptrdiff_t(dest_offset),  // dst_offset
```

Comment:
Why converting to `std::ptrdiff_t` explicitly here, but not for example in line 385 above? Just oversight or is there a different purpose?